checker.cpp

```diff
@@ -2,21 +2,57 @@
 #include <iostream>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
+class Battery{
+    public:
+        float temperature,soc,chargeRate;
+    bool checkTempIsInRange(float t);
+    bool checkSocIsInRange(float s);
+    bool checkChargeRateIsInRange(float c);
+    bool checkBatteryIsOk(float t,float s ,float c);
+};
+
+class NewBattery:public Battery
+{  
+    public:
+    bool checkTempIsInRange(float temperature)
+    {
+        if(temperature < 0 || temperature > 45)
```

Comment:
Similar range checking duplicate code can be reduced and refactored to a single method. for ex. IsValueInRange(...)

Reply:
Refactored and added a new method